libs/contrib/Data/Fun/Extra.idr

```diff
@@ -45,14 +45,14 @@ homoFunNeut_inv : id cod -> Fun [] cod
 homoFunNeut_inv x = x
 
 public export
-homoFunMult_inv : {rs : Vect n Type} -> (Fun rs . Fun ss) cod -> Fun (rs ++ ss) cod
+homoFunMult_inv : {n : Nat} -> {0 rs : Vect n Type} -> (Fun rs . Fun ss) cod -> Fun (rs ++ ss) cod
 homoFunMult_inv {rs = []     } gs = gs
 homoFunMult_inv {rs = t :: ts} fgs = \x => homoFunMult_inv (fgs x)
 
 
 ||| Apply an n-ary function to an n-ary tuple of inputs
 public export
-applyPartially : {ts : Vect n Type} -> {ss : Vect m Type} 
+applyPartially : {n : Nat} -> {0 ts : Vect n Type} 
```

Comment:
That's funny: the `HVect` argument gives you enough of a handle
on `n` to not need it but because the function is defined by
composing `homoFunMult_ext` we cannot remove it!

Note that `cod` could be made `0` though.

Reply:
It is funny :).
Someone showed me another instance of this phenomena in more 'traditional' mathematics:
|x|² = x²
but if we're only deducing differentiability compositionally, the LHS is non-differentiable at 0 while the RHS is.

Reply:
I should give credit where it's due, that someone was @jasigal